flair/models/lemmatization_model.py

```diff
@@ -0,0 +1,630 @@
+            self.char_embedding = nn.Embedding(len(self.character_dictionary), self.embeddings.embedding_length)
+            self.char_embedding.weight.data.copy_(self.pre_embedding_weight)
+            self.char_embedding.weight.requires_grad = False
+        else:
+            # Contextualized word embedding.
+            representation = self.embeddings.lm.get_representation(" ", start_token, end_token)
+            # Get the representation of the start sign in the contextualized word embedding.
+            self.start_representation = representation[:len(start_token)].detach()
+
+    def _load_embedding_weight(self, dictionary: Dictionary, embeddings: FlairEmbeddings):
+        """
+        According to character_dictionary, load weights from pre-trained Embedding.
+        When using classical word embedding, use this method to load the weights of the embedding layer.
+        """
+        loaded_embeddings = torch.zeros((len(dictionary), embeddings.embedding_length)).to(flair.device)
+        for i in range(0, len(dictionary)):
+            sentence = Sentence(dictionary.get_item_for_index(i))
+            if len(sentence) is not 0:
+                embeddings.embed(sentence)
+                loaded_embeddings[i] = sentence[0].embedding
+
+        return loaded_embeddings
+
+    def forward_loss(
+            self, data_points: Union[List[Sentence], Sentence], sort=True
+    ) -> torch.tensor:
+        loss = self.forward(data_points)
+
+        return loss
+
+    def forward(self, sentences: List[Sentence]):
+
+        loss = 0
+        for sentence in sentences:
+            encoder_input, lemma, mask, effective_data_lenght, max_lemma_lenght = self._generate_input(sentence)
+
+            encoder_out, encoder_hidden = self._encode(encoder_input, effective_data_lenght)
+
+            decoder_hidden = encoder_hidden[:self.n_layers]
+
+            # Using a contextualized word embedding model, it is necessary to generate a representation for the current moment
+            # based on the previously predicted results and context. The dictionary all_seqs is used to store the previous prediction results.
+            if self.contextualized_embedding:
+                batch_size = len(effective_data_lenght)
+                all_seqs = dict()
+                for i in range(batch_size):
+                    all_seqs[i] = ""
+            else:
+                all_seqs = None
+
+            # If you use teacher forcing, the next input uses the correct value, otherwise use the predicted value as input for the next moment.
+            # The probability of using Teacher Forcing can be modified by adjusting the value of teacher_forcing_ratio.
+            use_teacher_forcing = True if random.random() < self.teacher_forcing_ratio else False
+
+            decoder_input = None
+            if use_teacher_forcing:
+                for t in range(max_lemma_lenght):
```

Comment:
This if-clause could be placed only around the code that generates `decoder_input` since the rest is (I think) the same

Reply:
Yes, I've refactored the code here.